### DISCUSSION TITLE
Documentation for GitHub Codespaces

### DISCUSSION THREAD

**Discussion by samruddhikhandale** [withastro/roadmap#472] [Proposal]
### Body

Hi 👋 

I am Samruddhi, a Software Engineer for GitHub, working with the [GitHub Codespaces](https://github.com/features/codespaces) team, and a maintainer for the [dev containers](https://github.com/devcontainers) org 👩‍💻

I was looking at the [withastro/astro](https://github.com/withastro/astro/tree/main/.devcontainer) repo which has a pretty cool [dev container configuration](https://github.com/withastro/astro/blob/main/.Dockerfile) setup ✨ 

[GitHub Codespaces](https://docs.github.com/en/codespaces/overview) is a cloud-based development environment that uses your dev container to provide great contributor and maintainer experiences for all users of your project 🪄

🙅‍♀️ No one would ever have to worry about setting up their local machines, adding tools/runtimes for a specific project (which most frequently contradicts with their current environment) and spend hours before even starting their actual work 🙅‍♀️

Would you be willing to accept improvements to your current [dev container config](https://github.com/withastro/astro/blob/main/.devcontainer/devcontainer.json) and a documentation to improve the GitHub Codespaces experience? 💁‍♀️

Thanks,
Samruddhi

// cc @craiglpeters

**Comment by tony-sull** [comment]
Hey @samruddhikhandale! Thanks for reaching out, we'd be happy to review any improvements you have in mind for Codespaces support in Astro ✨  Is there anything we can do on our end to help?

I don't personally have much experience with web container configuration but I know we have quite a few core team members and maintainers that have worked with them quite a bit!

**astro.new**

As far as adding Codespaces links to [astro.new](https://astro.new), the main technical hurdle we've had in the past is related to our official starters living in the [withastro/astro](https://github.com/withastro/astro) monorepo.

With Codespaces will it be possible to open the container scoped to a specific directory in the monorepo?

**Comment by samruddhikhandale** [comment]
Thanks for your response, and for offering help. Much appreciated!

> With Codespaces will it be possible to open the container scoped to a specific directory in the monorepo?

Codespaces support monorepo scenarios, so I think that should be possible.

Tagging @bdmac who will be working on contributing this PR.

**Comment by bdmac** [comment]
I just put up a draft PR for this in https://github.com/withastro/astro/pull/6288
